aten/src/ATen/native/cpu/LogAddExp.h

```diff
@@ -7,14 +7,14 @@ namespace at { namespace native {
 inline namespace CPU_CAPABILITY {
 
 // custom min and max to be used in logcumsumexp for complex arguments
-template <typename scalar_t, bool min>
-c10::complex<scalar_t> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y) {
+template <typename scalar_t>
+std::pair<c10::complex<scalar_t>, c10::complex<scalar_t>> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y) {
   if (at::_isnan(y)) {  // either real is nan or imag is nan
-    return y;
+    return std::make_pair(y, y);
   } else if (at::_isnan(x)) {  // either real is nan or imag is nan
-    return x;
+    return std::make_pair(x, x);
```

Comment:
Previous code calls this function like below:
```
auto min = _logcumsumexp_minmax<scalar_t, true>(x, y);
auto max = _logcumsumexp_minmax<scalar_t, false>(x, y);
```
So if `y` has nans (regardless whether `x` has nans or not), min and max both are `y`. If `y` does not have nans, but `x` does, min and max both are `x`. Am I missing something here?

Reply:
Nope, you are right! I was thinking that perhaps this function is not as equivalent to its real counterpart as we would like (it returns (y, y) by default rather than, say, (y,x) or smth), but the thing is that, if one of the inputs is NaN, then we'll return a complex(NaN,NaN), so it doesn't really matter.